working/declaring-constructors/feature-specification.md

```diff
@@ -409,8 +595,31 @@ become a declaring one:
     declaring? Probably the one with the most initializing formals, but what do
     you do in case there's a tie?
 
+### Lint for `final` parameters
+
+This proposal retcons the `final` modifier on parameters to mean something
+specific in a declaring constructor. Outside of a declaring constructor the
+modifier is allowed but has a different effect: it simply makes the parameter
+itself non-assignable.
+
+If declaring constructors become popular, then users will likely start to see
+`final` before a parameter and read it as a field parameter. They will then be
+confused if the parameter isn't actually in a declaring constructor and isn't
+a field parameter.
+
+Given that non-assignable parameters aren't actually that *useful*, it may be
+worth discouraging users from marking a parameter with `final` unless it is a
+field parameter. This seems like a good candidate for a lint.
```

Comment:
If nobody writes `final` in front of parameters today, there is no problem.
If they do, it's because they want the parameter to be a final variable (because they like final variables), and then they'll probably get annoyed if they're told not to.

That measn the lint wont' be a recommended lint. It's too divisive. 
Will _anybody_ then use such a lint?
(And will there be another lint to always put `final` on parameters, unless they are mutated inside the body, because people who like `final` got reminded that they forgot that parameters are local variables?)

Not sure there is an issue at all. I'd definitely wait and see if anyone *asks* for a lint, and if so, which one.